src/main/java/seedu/teachstack/logic/commands/CommandResult.java

```diff
@@ -19,21 +22,38 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    private final boolean showPopUp;
+    private final Node additionalComponent;
+
     /**
      * Constructs a {@code CommandResult} with the specified fields.
      */
+    public CommandResult(String feedbackToUser, boolean showHelp, boolean exit,
+                         boolean showPopUp, Node additionalComponent) {
+        this.feedbackToUser = requireNonNull(feedbackToUser);
+        this.showHelp = showHelp;
+        this.exit = exit;
+        this.showPopUp = showPopUp;
+        this.additionalComponent = additionalComponent; // Assign the additional UI component
+    }
+
+    /**
+     * Constructs a {@code CommandResult} with specified fields
+     */
     public CommandResult(String feedbackToUser, boolean showHelp, boolean exit) {
         this.feedbackToUser = requireNonNull(feedbackToUser);
         this.showHelp = showHelp;
         this.exit = exit;
+        this.showPopUp = false; // Initialize showPopUp variable
+        this.additionalComponent = null;
     }
 
     /**
      * Constructs a {@code CommandResult} with the specified {@code feedbackToUser},
      * and other fields set to their default value.
      */
     public CommandResult(String feedbackToUser) {
-        this(feedbackToUser, false, false);
+        this(feedbackToUser, false, false, false, null);
```

Comment:
I was wondering if you need to update the test in CommandResultTest also, for the showPopUp and additionalComponent. 
I think thats probably the reason why you cant pass the codecov for this file?

Reply:
Noted